infra/nativetest/src/test/java/org/apache/shardingsphere/infra/nativetest/jdbc/features/ShadowTest.java

```diff
@@ -0,0 +1,149 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.nativetest.jdbc.features;
+
+import org.apache.shardingsphere.driver.api.yaml.YamlShardingSphereDataSourceFactory;
+import org.apache.shardingsphere.infra.nativetest.FileTestUtils;
+import org.apache.shardingsphere.infra.nativetest.jdbc.features.entity.Address;
+import org.apache.shardingsphere.infra.nativetest.jdbc.features.entity.Order;
+import org.apache.shardingsphere.infra.nativetest.jdbc.features.entity.OrderItem;
+import org.apache.shardingsphere.infra.nativetest.jdbc.features.repository.AddressRepository;
+import org.apache.shardingsphere.infra.nativetest.jdbc.features.repository.OrderItemRepository;
+import org.apache.shardingsphere.infra.nativetest.jdbc.features.repository.OrderRepository;
+import org.junit.jupiter.api.Test;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public final class ShadowTest {
+    
+    private OrderRepository orderRepository;
+    
+    private OrderItemRepository orderItemRepository;
+    
+    private AddressRepository addressRepository;
+    
+    @Test
+    void testShadowInLocalTransactions() throws SQLException, IOException {
+        DataSource dataSource = YamlShardingSphereDataSourceFactory.createDataSource(FileTestUtils.readFromFileURLString("yaml/shadow.yaml"));
+        orderRepository = new OrderRepository(dataSource);
+        orderItemRepository = new OrderItemRepository(dataSource);
+        addressRepository = new AddressRepository(dataSource);
+        this.initEnvironment();
+        this.processSuccess();
+        this.cleanEnvironment();
+    }
+    
+    private void initEnvironment() throws SQLException {
+        orderRepository.createTableIfNotExists();
+        orderItemRepository.createTableIfNotExists();
+        addressRepository.createTableIfNotExists();
+        orderRepository.truncateTable();
+        orderItemRepository.truncateTable();
+        addressRepository.truncateTable();
+        orderRepository.createTableIfNotExistsShadow();
+        orderRepository.truncateTableShadow();
+    }
+    
+    private void processSuccess() throws SQLException {
+        final List<Long> orderIds = insertData();
```

Comment:
If possible, collections (Set, List, ...) in ShardingSphere are declared as Collection<x>

Reply:
- @RaigorJiang Solved.